tests/behat/src/CmsFormsContext.php

```diff
@@ -310,11 +310,38 @@ protected function getHtmlField($locator)
     {
         $locator = $this->fixStepArgument($locator);
         $page = $this->getSession()->getPage();
+        
+        // Searching by name is usually good...
         $element = $page->find('css', 'textarea.htmleditor[name=\'' . $locator . '\']');
+        
+        if ($element === null) {
+            $element = $this->findInputByLabelContent($locator);
+        }
+        
         assertNotNull($element, sprintf('HTML field "%s" not found', $locator));
         return $element;
     }
 
+    protected function findInputByLabelContent($locator)
+    {
+        $page = $this->getSession()->getPage();
+        $label = $page->findAll('xpath', sprintf('//label[contains(text(), \'%s\')]', $locator));
+
+        if (empty($label)) {
+            return null;
+        }
+
+        assertCount(1, $label, sprintf(
+            'Found more than one element containing the phrase "%s".',
+            $locator
```

Comment:
This seems restrictive. Can you just return the first match instead of failing?

Reply:
I considered that but:
- I thought it would just be confusing for the implementor if it was matching the wrong field, doing something like typing into it and your test was failing
- If a new field is added the problem would me immediately apparent.
- The fix would probably be to make the match more specific anyway.